controllers/auth.go

```diff
@@ -55,56 +55,90 @@ func GoogleCallback(ctx *gin.Context, db *bun.DB) {
     domain := os.Getenv("DOMAIN")
     authRedirectUrl := os.Getenv("AUTH_REDIRECT_URL")
 
-    user := new(models.User)
     googleAccountInfo, getInfoError := getUserInfoFromCode(code, getGoogleOAuthConfig(), ctx)
-
     if getInfoError != nil {
-        log.Fatal(getInfoError)
-        ctx.JSON(500, gin.H{
-            "message": "error",
-        })
+        log.Printf("Failed to get user info: %v", getInfoError)
+        ctx.JSON(http.StatusInternalServerError, gin.H{"message": "error"})
         return
     }
 
-    count, err := db.NewSelect().Model(user).Where("email = ?", googleAccountInfo.Email).ScanAndCount(ctx)
+    log.Printf("Google account info: %v", googleAccountInfo)
+
+    var user models.User
 
+    tx, err := db.BeginTx(ctx, nil)
     if err != nil {
-        log.Fatal(err)
-        ctx.JSON(500, gin.H{
-            "message": "error",
-        })
+        log.Printf("Failed to begin transaction: %v", err)
+        ctx.JSON(http.StatusInternalServerError, gin.H{"message": "error"})
+        return
+    }
+
+    defer func() {
+        if p := recover(); p != nil {
+            tx.Rollback()
+            panic(p)
+        } else if err != nil {
+            tx.Rollback()
+        } else {
+            err = tx.Commit()
+        }
+    }()
```

Comment:
Are we doing this in a transaction?

Reply:
We are managing the database operations within a transaction